tests/run-make/rustc-crates-on-stable/rmake.rs

```diff
@@ -0,0 +1,42 @@
+//! Checks if selected rustc crates can be compiled on the stable channel (or a "simulation" of it).
+//! These crates are designed to be used by downstream users.
+
+use run_make_support::{cargo, run_in_tmpdir, rustc_path, source_root};
+
+fn main() {
+    run_in_tmpdir(|| {
+        // Use the stage0 beta cargo for the compilation (it shouldn't really matter which cargo we
+        // use)
+        let cargo = cargo()
+            // Ensure `proc-macro2`'s nightly detection is disabled
+            .env("RUSTC_STAGE", "0")
```

Comment:
Also, while this inhibits `proc-macro2`'s auto detection, some other crates may parse rustc's version, instead of using a build probe like `proc-macro2`, without a way to opt-out. 

It's not the case today, but we could also use the recent env var `RUSTC_OVERRIDE_VERSION_STRING` to masquerade as a stable release version. For example, just removing the "dev"/"nightly" suffixes from the output of stage1 `rustc --version` should work.

(It would not prevent `proc-macro2`'s probing by itself so we couldn't only use that -- but I thought I'd mention it as a future possibility)

Reply:
In a long run, this could cause some problems if the crates were detecting e.g. the specific version of stable and enabling some features based on that, I guess. Feel free to use the env. var., I think it's fine without it for now, but it shouldn't hurt. Up to you.

Reply:
There's no perfect solution really without rustc being able to functionally masquerade as another version, and we don't want to do that. If we find other crates that rely on the version string instead of build probes, we can use the env var in the future, it's made for that very purpose. Until then, only `proc-macro2` caused issues and we have it working already, so we should be good here for now.